packages/shared_preferences/shared_preferences_ios/lib/shared_preferences_ios.dart

```diff
@@ -2,52 +2,71 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:async';
-
 import 'package:flutter/services.dart';
 import 'package:shared_preferences_platform_interface/shared_preferences_platform_interface.dart';
+import 'messages.g.dart';
+
+typedef _Setter = Future<void> Function(String key, Object value);
 
-const MethodChannel _kChannel =
-    MethodChannel('plugins.flutter.io/shared_preferences_ios');
+/// iOS implementation of shared_preferences.
+class SharedPreferencesIos extends SharedPreferencesStorePlatform {
+  final UserDefaultsApi _api = UserDefaultsApi();
+  late final Map<String, _Setter> _setters = <String, _Setter>{
+    'Bool': (String key, Object value) {
+      return _api.setBool(key, value as bool);
+    },
+    'Double': (String key, Object value) {
+      return _api.setDouble(key, value as double);
+    },
+    'Int': (String key, Object value) {
+      return _api.setValue(key, value as int);
+    },
+    'String': (String key, Object value) {
+      return _api.setValue(key, value as String);
+    },
+    'StringList': (String key, Object value) {
+      return _api.setValue(key, value as List<String?>);
+    },
+  };
 
-/// The macOS implementation of [SharedPreferencesStorePlatform].
-///
-/// This class implements the `package:shared_preferences` functionality for iOS.
-class SharedPreferencesIOS extends SharedPreferencesStorePlatform {
-  /// Registers this class as the default instance of [SharedPreferencesStorePlatform].
+  /// Registers this class as the default instance of [PathProviderPlatform].
   static void registerWith() {
-    SharedPreferencesStorePlatform.instance = SharedPreferencesIOS();
+    SharedPreferencesStorePlatform.instance = SharedPreferencesIos();
   }
 
   @override
-  Future<bool> remove(String key) async {
-    return (await _kChannel.invokeMethod<bool>(
-      'remove',
-      <String, dynamic>{'key': key},
-    ))!;
+  Future<bool> clear() async {
+    final Map<String, Object> all = await getAll();
+    for (final String key in all.keys) {
+      await _api.remove(key);
```

Comment:
Yea, I wanted to take the code to the logical extreme of what we are saying we wanted to see how it looked.  If our goal is to move logic from host to dart that's often going to incur a performance cost.

I moved it back to the native side.

Reply:
Fortunately logical extremes are not the only options available to us. As with all code, we can balance performance against other design goals, favoring the general goal in the general case, and adjusting for performance in specific cases where it will obviously have a significant impact (like here, where moving a few lines of code puts a potentially large number of cross-boundary calls in a tight loop).